doc/srv6/srv6-hld-v19.md

```diff
@@ -0,0 +1,552 @@
+
+In SRv6 domain,  TE policy associated with SID list could be configured on headend nodes, to steer traffic with SRH encapsulation. When traffic reaches egress nodes,  the packets are processed based on local defined functions, for example SID list decapsulation and FIB lookup in a particular VRF . 
+
+# 2 Feature Requirements
+
+## 1.1 Functional Requirements
+
+This section describes the SONiC requirements for SRv6 feature in phases:
+
+At a high level the following should be supported:
+
+Phase #1
+
+​	Should be able to perform the role of SRv6 domain headend node, and endpoint node, more specific:
+
+- Support H.Encaps, H.Encaps.Red, END.B6.Encaps, END.B6.Encaps.Red
+- Support END, END.X
+- Support END.DT46
+- Support traffic steering on SID list
+- Support anycast SID
+
+Later phases:
+
+- Support uSID/G-SID
+- Other programming functions
+- Support HMAC option
+- Support sBFD for SRv6 
+
+This document will focus on Phase #1, while keep the design extendable for future development
+
+## 1.2 Configuration and Management Requirements
+
+1. User should be able to enable SRv6 globally or per interface
+
+2. User should be able to configure SID list for encapsulation
+
+3. User should be able to configure SRv6 steering policy
+
+4. User should be able to configure endpoint action and corresponding argument for matched local SID
+
+## 1.3 Scalability Requirements
+
+- Maximum number of end.DT46 SID: 2k
+- Maximum number  of Binding SID: 1k
+- Maximum number of SR policy: 1k
+- Maximum SID-list per SR Policy: 4
+- Maximum number of SID-list: 4k
+- SRH depth with regular SID: 3
+
+## 1.4 Warm Boot Requirements
+
+Warm reboot is intended to be supported for planned system, swss and BGP warm reboot.
+
+ 
+
+# 2 Feature Design
+
+![draw-configdb](images/drawing-configdb-frr3.png)
+
+Before FRR is ready, we will use static configuration to set SIDs and apply policy for TE. It enables basic SRv6 operation and populates SRv6 into ASIC, allows SRv6 data plan forwarding. More complicated SRv6 policy can be enabled when SRv6 is fully supported in FRR and passed from FRR to fpmsyncd.
+
+## 2.1 ConfigDB Changes
+
+**SRV6_SID_LIST_TABLE**
+
+Description: New table that stores SRv6 SID list configuration.
+
+Schema:
+
+```
+; New table
+; holds SRv6 SID list
+
+key = SRV6_SID_LIST:segment_name
+                     ; SID segment name
+; field = value
+path = SID,          ; List of SIDs
+
+For example:
+    "SRV6_SID_LIST": {
+        "seg1": {
+            "path": [
+                "BABA:1001:0:10::",
+                "BABA:1001:0:20:F1::"
+            ]
+        },
+        "seg2": {
+            "path": [
+                "BABA:1001:0:30::",
+                "BABA:1001:0:40:F1::"
+            ]
+        }
+    }
+```
+
+**SRV6_LOCAL_SID_TABLE**
+
+Description: New table to hold local SID to behavior mapping
+
+Schema:
+
+```
+; New table
+; holds local SID to behavior mapping, allow 1:1 or n:1 mapping
+
+key = SRV6_LOCAL_SID:ipv6address
```

Comment:
config-DB table should use '|' key seperator.

Reply:
fixed

Reply:
I do not see this addressed. Please fix for all tables in this section